### ISSUE TITLE
"An error occurred during the publish operation" on Amplify Publish (Linux)

### ISSUE THREAD

**Issue by rossbishop** [aws-amplify/amplify-cli#5990]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
Amplify publish appears to be broken at the very least on Ubuntu with my specific React project. Seems to work fine on Windows, where the app was originally created. App hosted in eu-west-2.

**Amplify CLI Version**
4.35.0, also tried 4.36.0 and 4.30.0

**To Reproduce**
1. Clone repo with amplify backend + frontend configuration
2. Run `amplify init`, configure aws profile if required
3. Run `amplify pull` to sync up cloud with local 
4. Run `amplify publish`
5. Backend changes should be fine, after string `Publish started for amplifyhosting` error returns:
`command execution terminated with error
An error occurred during the publish operation`

**Expected behavior**
Amplify publish should work
When an error occurs, more than 
`command execution terminated with error
An error occurred during the publish operation` should be returned. A stack trace would be extremely useful.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 (tried with WSL, GitHub Actions, desktop), 20.04 (tried with Github Actions)
 - Node Version(s): 12.x, 15.x

**Additional context**
This seems like it might be a build issue or similar, though I can get an `npm react-scripts build` to work in isolation, providing react-scripts is installed. Been fighting with this issue for a week solid and can't find a way out of it.

**Comment by kaustavghosh06** [comment]
@rossbishop Can you please check your `amplify/.config/project-config` and see all the build commands? If you would like to change those commands from linux->windows specifically, you'll have to run `amplify configure project` or manually update that file if you see a mismatch there.

**Comment by rossbishop** [comment]
AH! This might well be it. Publish just succeeded on my WSL setup. I hadn't spotted that detail, I've tried all sorts in the past week. So for whatever reason by default I guess when a project is created on windows the build command is "npm.cmd", but so far as I can tell, the ".cmd" suffix isn't required, as I just removed it from my local Windows copy of the repo and publish worked fine.

What's the reasoning for having this in there by default?
